### PR TITLE
Make sure to only include what is necessary

### PR DESCRIPTION
**Pull request by linkmauve**
And thank you, include-what-you-use!

### CODE REVIEW

File: include/colors.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef _COLORS_H
 #define _COLORS_H
 
-#include <termbox.h>
```
Comment (SirCmpwn) [inline include/colors.h]: :-1:

File: include/imap/date.h [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #ifndef _IMAP_DATE_H
 #define _IMAP_DATE_H
 
-#include <time.h>
+struct tm;
```
Comment (SirCmpwn) [inline include/imap/date.h]: :-1: again

File: include/state.h [hunk 3]
```diff
@@ -3,8 +3,11 @@
 
 #include <pthread.h>
 #include <time.h>
+
 #include "util/list.h"
-#include "worker.h"
+
+struct aerc_mailbox;
+struct aerc_message;
```
Comment (SirCmpwn) [inline include/state.h]: And so on

**Comment by linkmauve** [comment]
This is the only way to make headers independent of their order of inclusion.

**Comment by SirCmpwn** [comment]
:+1: unless otherwise noted

And that's fine, I would rather include the headers that have "ownership" over those structures.

**Comment by linkmauve** [comment]
Alright, reverted that part.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**